include/swift/AST/DiagnosticsSema.def

```diff
@@ -1735,7 +1735,10 @@ ERROR(functions_mutating_and_not,none,
       "method must not be declared both %0 and %1",
       (SelfAccessKind, SelfAccessKind))
 ERROR(static_functions_not_mutating,none,
-      "static functions must not be declared mutating", ())
+      "%0 modifier cannot be used on static functions",
+      (SelfAccessKind))
+NOTE(static_functions_not_mutating_detail,none,
+      "static members cannot access instance properties or self", ())
```

Comment:
I’m not a fan of this interpretation because it’s misleading. We have this very problem with another diagnostic: https://github.com/swiftlang/swift/issues/62909. You can access both `self` or instance members inside a static method depending on what you mean by it.

Maybe we could borrow a more clear phrasing from the [book](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/methods#Type-Methods). For example:

```
static members are defined on the type itself rather than on instances of the type
static members belong to the type itself rather than to instances of the type
static members are members of the type itself rather than instances of the type
```